api_v3/lib/KalturaAssetService.php

```diff
@@ -88,9 +88,22 @@ protected function serveAsset(asset $asset, $fileName, $forceProxy = false, $ver
 			
 			case StorageProfile::STORAGE_SERVE_PRIORITY_KALTURA_ONLY:
 				$fileSync = kFileSyncUtils::getReadyInternalFileSyncForKey($syncKey);
-				if(!$fileSync)
+				if($fileSync)
+				{
+					break;
+				}
+
+				$periodicStorageIds = kStorageExporter::getPeriodicStorageIdsByPartner($asset->getPartnerId());
```

Comment:
This entire block of code - move to remoteStorage class (we will reuse it)
If (!fileSync)
{
   fileSync= remoteStorage::getGlobalReadyFileSync(syncKey) //we might have a better name later
  if(!fileSync) throw.
}